src/Microsoft.Azure.Functions.Extensions.Mcp/Trigger/McpToolTriggerRichContentReturnValueBinder.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Text.Json;
+using Microsoft.Azure.Functions.Extensions.Mcp.Serialization;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using ModelContextProtocol.Protocol;
+
+namespace Microsoft.Azure.Functions.Extensions.Mcp;
+
+internal class McpToolTriggerRichContentReturnValueBinder(CallToolExecutionContext executionContext) : IValueBinder
+{
+    public Type Type { get; } = typeof(object);
+
+    public Task SetValueAsync(object value, CancellationToken cancellationToken)
+    {
+        if (value is null)
+        {
+            executionContext.SetResult(null!);
+            return Task.CompletedTask;
+        }
+
+        if (value is not string jsonString)
+        {
+            throw new ArgumentException("Expected JSON string.", nameof(value));
+        }
+
+        var mcpToolResult = JsonSerializer.Deserialize<McpToolResult>(jsonString, McpJsonSerializerOptions.DefaultOptions)
+                    ?? throw new InvalidOperationException("The function return value could not be deserialized to a valid McpToolResult.");
+
+        if (mcpToolResult.Content is null)
+        {
+            throw new ArgumentNullException(mcpToolResult.Content, "The McpToolResult content cannot be null.");
+        }
+
+        ContentBlock? content;
```

Comment:
Content here should be a collection to enable multiple results (e.g., different content types, batch results, multi-output formats, etc.). That will match the expectation from `CallToolResult` (you won't have to create a single element collection when this is done).

The logic here would need to treat the raw content as a potential collection. You should consider making that explicit in the contract, where the `Type` would indicate multiple content entries (e.g., `multi-content-result`). That means you'd be able to keep tour existing handling for the types you support (`audio`, `image`, etc.), but you'd need to be able to reuse this when handling a collection of content items, which is a good change as it would also simplify/reduce some of the mapping code.

Reply:
Addressed this feedback, let me know what you think!